### ISSUE TITLE
feat: support initializing with Bun

### ISSUE THREAD

**Issue by Zeko369** [t3-oss/create-t3-app#1366]
### Provide environment information

  System:
    OS: macOS 13.4
    CPU: (8) arm64 Apple M1
    Memory: 102.36 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.18.0 - ~/Library/Caches/fnm_multishells/78864_1681854261277/bin/node
    npm: 8.19.2 - ~/Library/Caches/fnm_multishells/78864_1681854261277/bin/npm
    Watchman: 2023.01.02.00 - /opt/homebrew/bin/watchman

### Describe the bug

If running create-t3-app with bunx it defaults to installing depends with `npm` rather than `bun install`

### To reproduce

`bunx create-t3-app`

Should prompt for "install depends with bun" instead of npm

### Additional information

I have a patch ready if you're ok with this change, the performance diff is massive...

![CleanShot 2023-04-19 at 00 13 20](https://user-images.githubusercontent.com/3064377/232916770-a2871ace-2556-48b4-9f34-d57f2e72a271.png)

**Comment by Zeko369** [comment]
> Bun cannot currently be used as an end to end solution. You can use it to init and install dependencies, but as far as I can tell if you try to run the dev server with it, it will start it but fail to serve requests correctly.

Yes you can't use custom `bun dev` server to run a next app (at least not fully and not yet)

But it's the same as with `pnpm`, (you're not running the app with pnpm it's running with node, just being called from pnpm), while bun has a dev server (`bun dev` that's not yet stable) running `bun run dev` is the same as `npm run dev` / `pnpm run dev` / `yarn dev`

Also using tools like [ni](https://github.com/antfu/ni) the `nr` commend is an alias for `npm run` / `pnpm run` / `bun run`

*TLDR; While Bun has a WIP dev server it can be used exactly as pnpm*
- `pnpm install` -> `bun install`
- `pnpm dlx` -> `bunx`
- `pnpm run dev` -> `bun run dev`

**Comment by yaikohi** [comment]
One of the other reasons why you cannot run t3-app with `bun` is because `prisma` doesn't work with `bun` yet.
There is an issue at oven-sh/bun where the `prisma` compatibility is being tracked, so I am just going to link that here.

https://github.com/oven-sh/bun/issues/2083

**Comment by Zeko369** [comment]
I know prisma doesn't work / also next itself has a fork with a subset of features that works under bun, BUT that's not the point of this issue / PR

Bun isn't just a `node`/`deno` replacement, it has a super fast package manager you can use with any node project (just like you don't run your code with `pnpm` you install your packages with it, you can use bun to just install packages)

https://bun.sh/docs/cli/install
https://twitter.com/jarredsumner/status/1606163655527059458
